otel.go

```diff
@@ -34,12 +34,22 @@ func (processor *attributeSpanProcessor) Shutdown(context.Context) error { retur
 // ForceFlush is a noop to satisfy the SpanProcessor interface.
 func (processor *attributeSpanProcessor) ForceFlush(context.Context) error { return nil }
 
-// ContextWithAttributes returns a child of the ctx containing the attributes. All spans
-// created with the returned context and its children will have the attributes set.
+// ContextWithAttributes returns a child of the ctx containing the attributes.
+// All spans created with the returned context and its children will have the
+// attributes set. If there are any existing attributes on the context already,
+// the returned child context will override all of the existing attributes.
 func ContextWithAttributes(ctx context.Context, attributes []attribute.KeyValue) context.Context {
```

Comment:
is there a reason we wouldnt want to always use ContextWithAppendedAttributes? should we delete this function?

Reply:
While I don't think we currently have a use for it, I don't think we should remove the old one because 1. it requires updating a bunch of libraries to fix the backward-incompatible change and 2. it's useful to have some flexibility in helper functions so if we do want to remove/overwrite attributes in the future, it's pretty easy. If we had only append, we wouldn't be able to change values for existing attributes that were previously set.